### DISCUSSION TITLE
Eslint shows errors `no-unused-vars` and `no-undef` when declare ENV with `typing.d.ts` in react app

### DISCUSSION THREAD

**Discussion by nqhd3v** [eslint/eslint#16133] [Q&A] [answered]
Node version: 16.15.1
npm version: 8.11.0
Local ESLint version:  8.17
Operating System: macOS

----
Hi everyone, in my reactJS source, I used `typing.d.ts` to declare module and env variables in my project.
But after I init eslint, it shows errors `no-unused-vars` and `no-undef`.
Here is my `typing.d.ts` file:  
<img width="749" alt="image" src="https://user-images.githubusercontent.com/106160187/179135097-162159a8-260d-4730-af76-d57acf50b8e2.png">

File used that variables:  
<img width="661" alt="image" src="https://user-images.githubusercontent.com/106160187/179135230-2dca40a7-c990-4cca-8eaf-4f0b13f7e98c.png">

This is my `.eslintrc.js`:  
<img width="400" alt="image" src="https://user-images.githubusercontent.com/106160187/179135784-5211efe7-1f2c-4453-9753-f0551076968b.png">

So, how to fix it!?

**Comment by aladdin-add** [comment, marked as answer]
it's because airbnb does not support ts. if using ts, you should be able to replace `no-unused-vars` to `@typescript-eslint/no-unused-vars` likewise.

the easiest way to fix it is to extend `@typescript-eslint/recommended`.

```js
extends: [...., "plugin:@typescript-eslint/recommended"]
```

**Reply by nqhd3v** [reply]
Thank you so much! So it's because Airbnb doesn't support typescript!
Anyways, I think it will be worked, but, I will try to find another standard library, which supports typescript!
